include/aws/crt/iot/DeviceDefender.h

```diff
@@ -0,0 +1,200 @@
+                DeviceDefenderV1ReportTaskConfigBuilder &WithDeviceDefenderReportFormat(
+                    DeviceDefenderReportFormat reportFormat) noexcept;
+
+                /**
+                 * Sets the task period nanoseconds. Defaults to 5 minutes.
+                 */
+                DeviceDefenderV1ReportTaskConfigBuilder &WithTaskPeriodNs(uint64_t taskPeriodNs) noexcept;
+
+                /**
+                 * Sets the network connection sample period nanoseconds. Defaults to 5 minutes.
+                 */
+                DeviceDefenderV1ReportTaskConfigBuilder &WithNetworkConnectionSamplePeriodNs(
+                    uint64_t networkConnectionSamplePeriodNs) noexcept;
+
+                /**
+                 * Sets the task cancelled handler function.
+                 */
+                DeviceDefenderV1ReportTaskConfigBuilder &WithDefenderV1TaskCancelledHandler(
+                    OnDefenderV1TaskCancelledHandler &&onCancelled) noexcept;
+
+                /**
+                 * Sets the user data for the task cancelled handler function.
+                 */
+                DeviceDefenderV1ReportTaskConfigBuilder &WithDefenderV1TaskCancellationUserData(
+                    void *cancellationUserdata) noexcept;
+
+                /**
+                 * Builds a device defender v1 task configuration object from the set options.
+                 */
+                DeviceDefenderV1ReportTaskConfig Build() noexcept;
+
+                /**
+                 * @return the value of the last aws error encountered by operations on this instance.
+                 */
+                int LastError() const noexcept { return aws_last_error(); }
+
+              private:
+                std::shared_ptr<Mqtt::MqttConnection> m_mqttConnection;
+                ByteCursor m_thingName;
+                Io::EventLoopGroup m_eventLoopGroup;
+                DeviceDefenderReportFormat m_reportFormat;
+                uint64_t m_taskPeriodNs;
+                uint64_t m_networkConnectionSamplePeriodNs;
+                OnDefenderV1TaskCancelledHandler m_onCancelled;
+                void *m_cancellationUserdata;
+            };
+
+            /**
+             * Represents a persistent DeviceDefender V1 task.
+             */
+            class AWS_CRT_CPP_API DeviceDefenderV1ReportTask final
+            {
+              public:
+                DeviceDefenderV1ReportTask(
+                    Aws::Crt::Allocator *allocator,
+                    const DeviceDefenderV1ReportTaskConfig &config) noexcept;
+                ~DeviceDefenderV1ReportTask();
+                DeviceDefenderV1ReportTask(const DeviceDefenderV1ReportTask &) = delete;
+                DeviceDefenderV1ReportTask(DeviceDefenderV1ReportTask &&) = delete;
+                DeviceDefenderV1ReportTask &operator=(const DeviceDefenderV1ReportTask &) = delete;
+                DeviceDefenderV1ReportTask &operator=(DeviceDefenderV1ReportTask &&) = delete;
```

Comment:
Is this the move assignment operator? Would it offer value if we keep this? It am guessing the implementation is simple.

Reply:
Good point, it could be of use. I've added implementation for the move operator in updated commits.